api/validation.go

```diff
@@ -74,6 +84,42 @@ func (api *API) validateAndHydrateDimensions(v dataset.Version, dims []model.Dim
 	}
 }
 
+// Used for POST Dimension when the filter type is multivariate.
+// Extra logic required to hydrate dimension using the default categorisation of the dimension.
+func (api *API) HydrateMultivariateDimensions(dimensions []model.Dimension, pType string) ([]model.Dimension, error) {
```

Comment:
I don't like the naming and the comment.. HydrateMultiVariateDimensions suggests all multivariate hydration should be going through this but the comment (and my understanding) suggests is just the one off case for POST dimensions.